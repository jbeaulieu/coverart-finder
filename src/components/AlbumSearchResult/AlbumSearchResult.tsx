import { Avatar, ListItem, ListItemAvatar, ListItemButton, ListItemText } from "@mui/material";

export type Props = {
  id: number;
  selected: boolean;
  artistName: string;
  albumName: string;
  albumThumbSrc?: string;
  onSelect: (selected: number) => void;
}

const AlbumSearchResult = (props: Props) => {

  const { id, selected, artistName, albumName, albumThumbSrc, onSelect } = props;

  return (
    <ListItem>
      <ListItemButton selected={selected} onClick={() => onSelect(id)}>
        <ListItemText
          primary={artistName}
          secondary={albumName}
        />
        <ListItemAvatar>
          <Avatar alt={albumName} src={albumThumbSrc} />
        </ListItemAvatar>
      </ListItemButton>
    </ListItem>
  );
};

export default AlbumSearchResult;
