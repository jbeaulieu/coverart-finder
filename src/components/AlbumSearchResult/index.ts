import AlbumSearchResult from './AlbumSearchResult';

export default AlbumSearchResult;
