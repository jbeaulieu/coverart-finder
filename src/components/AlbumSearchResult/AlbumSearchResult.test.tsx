import { fireEvent, render, screen } from '@testing-library/react';
import { describe, expect, it, vi } from 'vitest';
import AlbumSearchResult from './AlbumSearchResult';
import type { Props } from './AlbumSearchResult';

const albumId = 1767673630;
const artistName = 'Boys Like Girls';
const albumName = 'Watch the Fire';
const onSelect = vi.fn();

const defaultProps: Props = {
  id: albumId,
  selected: false,
  artistName,
  albumName,
  albumThumbSrc: undefined,
  onSelect
};

const renderSearchContainer = (props = defaultProps) => {
  return render(<AlbumSearchResult {...props} />);
};

describe('Component SearchContainer', () => {

  it('renders without crashing', () => {
    const { asFragment } = renderSearchContainer();

    expect(asFragment()).toMatchSnapshot();
  });

  describe('on click', () => {
    it('invokes onSelect function', () => {
      renderSearchContainer();

      const itemButton = screen.getByRole("button", { name: `${artistName} ${albumName}` });

      fireEvent.click(itemButton);

      expect(onSelect).toHaveBeenCalledExactlyOnceWith(albumId);
    });
  });
});
