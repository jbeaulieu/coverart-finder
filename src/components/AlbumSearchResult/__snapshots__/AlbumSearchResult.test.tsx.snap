// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Component SearchContainer > renders without crashing 1`] = `
<DocumentFragment>
  <li
    class="MuiListItem-root MuiListItem-gutters MuiListItem-padding MuiListItem-alignItemsFlexStart css-m0pems-MuiListItem-root"
  >
    <div
      class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-4bx69l-MuiButtonBase-root-MuiListItemButton-root"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiListItemText-root MuiListItemText-multiline css-14ln1j6-MuiListItemText-root"
      >
        <span
          class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-rizt0-MuiTypography-root"
        >
          Boys Like Girls
        </span>
        <p
          class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-1a1whku-MuiTypography-root"
        >
          Watch the Fire
        </p>
      </div>
      <div
        class="MuiListItemAvatar-root css-1ulznza-MuiListItemAvatar-root"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-18k2bs-MuiAvatar-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-1mo2pzk-MuiSvgIcon-root-MuiAvatar-fallback"
            data-testid="PersonIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
            />
          </svg>
        </div>
      </div>
    </div>
  </li>
</DocumentFragment>
`;
